src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -52,6 +52,15 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Queryable Entities
 
+(defn augment-for-subquery
+  [{:keys [entity] :as query-rec}]
+  (case entity
+    :facts (assoc-in query-rec
+             [:projections "value" :field]
+             (h/coalesce :fv.value_string
+                         (h/scast :fv.value_boolean :text)))
+    query-rec))
```

Comment:
I'm not wild about this, but the reason for it is that we allow queries like this:
```
curl -X GET http://localhost:8080/v4/events -d 'query=["in","new_value",["extract","value",["select_facts",["~","certname",".*"]]]]'
```

This presents a problem when the strings in the value column are serialized as json, since it becomes a comparison an escaped and nonescaped string.  This only comes into play when the facts query rec is a subquery, so I think this function leaves us on solid ground, but it's worth understanding the reasoning and thinking about alternatives. This makes no attempt to handle the case where new_value is numeric. That's not something we currently handle either.